astropy/units/core.py

```diff
@@ -485,7 +485,7 @@ class UnitsWarning(AstropyWarning):
     """
 
 
-class UnitBase(metaclass=InheritDocstrings):
+class UnitBase(metaclass=type):
```

Comment:
Removing `metaclass` broke CI before, but I found this seems to fix things. cc @mhvk

Reply:
Seems very weird that you could not just do `class UnitBase:` - what goes wrong?

Reply:
`__new__` complains about numbers of arguments are incorrect if I simply do `class UnitBase:` (which would inherit from `object`).

Reply:
Even now `UnitBase` inherits from `object` and I thought the default metaclass is `type` - which is why I am so confused that explicitly stating this makes any difference!

Reply:
I'll try to investigate locally...

Reply:
Okay, lemme know. I am not familiar with metaclasses. :woman_shrugging:

Reply:
Actually I cannot reproduce it: just using `class UnitBase:` works fine. Did you perhaps confuse with `UnitMeta`, which should inherit from `type` (as you have it).

Reply:
Hmm I'll double check. I plan to undo the `type` stuff where I can in a follow up commit anyway. I was just trying to get the CI to be green again first as a baseline. Thanks!

Reply:
OK, makes sense.

Reply:
You were right. I cleaned it up.